### PR TITLE
backport: bitcoin#18788, #18974, #19230, #19239, #19441, #19568, #19668, #20226, #21049, #22446 (descriptor wallets part II)

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
This PR continues supporting of Descriptor Wallets, see https://github.com/dashpay/dash/pull/5579 for prior work.
See https://github.com/dashpay/dash-issues/issues/59 for Check-list issue

## What was done?

Particularly:
 - bitcoin/bitcoin#18974
 - bitcoin/bitcoin#19230
 - bitcoin/bitcoin#19239
 - bitcoin/bitcoin#19441
 - bitcoin/bitcoin#19568
 - bitcoin/bitcoin#20226
 - bitcoin/bitcoin#21049
 - bitcoin/bitcoin#18788
 - bitcoin/bitcoin#22446

Beside that fixed:
 - drop requirement of HD in CanGetAddresses for watch-only wallets
 - unify ScriptPubKeyMan implementation for KeyOrigin with bitcoin's
 - minor dashification for  bitcoin/bitcoin#18067

## How Has This Been Tested?
Run unit + functional test. Backport bitcoin#18788 enables descriptor wallets for multiple functional tests.

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
Any objects to delaying this to post v20.1?

**Comment by knst** [comment]
> Any objects to delaying this to post v20.1?
no, it can't be merged before https://github.com/dashpay/dash/pull/5579 anyway

File: test/functional/wallet_importdescriptors.py [hunk 1]
```diff
@@ -98,6 +98,15 @@ def run_test(self):
...
         assert_equal(w1.getwalletinfo()['keypoolsize'], 0)
 
+        # Check persistence of data and that loading works correctly
+        w1.unloadwallet()
+        self.nodes[1].loadwallet('w1')
+        test_address(w1,
+                     key.p2pkh_addr,
+                     solvable=True,
+                     ismine=True,
+                     labels=["Descriptor import test"])
```
Comment (PastaPastaPasta) [inline test/functional/wallet_importdescriptors.py:109]: why is this ordering different?

**Review by PastaPastaPasta — COMMENTED** [review]
I think looks good overall; one question

File: test/functional/wallet_importdescriptors.py (hunk 1 above)
Comment (knst) [reply test/functional/wallet_importdescriptors.py:109]: The mine purpose is import+test+unload+load and retest address again.

We don't have required code snippet due to segwit. DNM. So, I re-implemented that test (unload / load & test) for another imported descriptor.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK 85fa6e4585c58e6b24a04ec2844cf671d80116b7

**Pull request merged**